test/functional/terminal/buffer_spec.lua

```diff
@@ -224,6 +225,23 @@ describe(':terminal buffer', function()
       neq('terminal', eval('&buftype'))
     end)
   end)
+
+  it('it works with set rightleft #11438', function()
+    local columns = eval('&columns')
+    feed(string.rep('a', columns))
+    command('set rightleft')
+    wait()
+    screen:expect([[
+                                               ydaer ytt|
+      {1:a}aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa|
+                                                        |
+                                                        |
+                                                        |
+                                                        |
+      {3:-- TERMINAL --}                                    |
+    ]])
+    feed_command('bdelete!')
```

Comment:
`feed_command` uses `feed()` (AKA `nvim_input`) so it is asynchronous. Better to use `command` usually.
```suggestion
    command('bdelete!')
```